src/modules/flow/json/json.c

```diff
@@ -224,4 +239,219 @@ json_object_get_key_in_process(struct sol_flow_node *node, void *data, uint16_t
+
+    return 0;
+}
+
+static void
+json_array_get_index_close(struct sol_flow_node *node, void *data)
+{
+    struct sol_json_array_index *mdata = data;
+
+    if (mdata->json_array)
+        sol_blob_unref(mdata->json_array);
+}
+
+static int
+json_array_index_process(struct sol_flow_node *node, struct sol_json_array_index *mdata)
+{
+    struct sol_json_scanner scanner;
+    enum sol_json_loop_reason reason;
+    struct sol_json_token token;
+
+    if (mdata->index < 0 || !mdata->json_array)
+        return 0;
+
+    sol_json_scanner_init(&scanner, mdata->json_array->mem,
+        mdata->json_array->size);
+
+    if (json_array_get_at_index(&scanner, &token, mdata->index, &reason))
+        return send_token_packet(node, &scanner, mdata->json_array, &token);
+
+    if (reason == SOL_JSON_LOOP_REASON_INVALID)
+        return sol_flow_send_error_packet(node, EINVAL,
+            "Invalid JSON array (%.*s)", (int)mdata->json_array->size,
+            (char *)mdata->json_array->mem);
+
+    return sol_flow_send_error_packet(node, EINVAL,
+        "JSON array index out of bounds: %d", mdata->index);
+}
+
+static int
+json_array_get_index_index_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct sol_json_array_index *mdata = data;
+    int r;
+    struct sol_irange in_value;
+
+    r = sol_flow_packet_get_irange(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (in_value.val < 0)
+        return sol_flow_send_error_packet(node, EINVAL,
+            "Invalid negative JSON array index: %d", in_value.val);
+
+    mdata->index = in_value.val;
+    return json_array_index_process(node, mdata);
+}
+
+static int
+json_array_get_index_in_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct sol_json_array_index *mdata = data;
+    int r;
+    struct sol_blob *in_value;
+
+    r = sol_flow_packet_get_json_array(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (mdata->json_array)
+        sol_blob_unref(mdata->json_array);
+    mdata->json_array = sol_blob_ref(in_value);
+    SOL_NULL_CHECK(mdata->json_array, -errno);
+
+    return json_array_index_process(node, mdata);
+}
+
+static int
+json_array_length_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_blob *in_value;
+    struct sol_json_scanner scanner;
+    enum sol_json_loop_reason reason;
+    struct sol_json_token token;
+    struct sol_irange len = { 0, 0, INT32_MAX, 1 };
+
+    r = sol_flow_packet_get_json_array(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_json_scanner_init(&scanner, in_value->mem, in_value->size);
+    SOL_JSON_SCANNER_ARRAY_LOOP_ALL(&scanner, &token, reason) {
+        if (!sol_json_scanner_skip_over(&scanner, &token))
+            return -EINVAL;
+
+        if (len.val == INT32_MAX)
```

Comment:
nice you did this here, but please add this check to the object length as well :-)